src/gui/abstract_menu.py

```diff
@@ -0,0 +1,125 @@
+import sys
+import pygame
+from src.settings import SCREEN_WIDTH, SCREEN_HEIGHT
+from src.support import resource_path
+from src.gui.components import Button
+from src.enums import GameState
+from pygame.mouse import get_pressed as mouse_buttons
+from pygame.math import Vector2 as vector
+from abc import ABC, abstractmethod
+
+_SCREEN_CENTER = (SCREEN_WIDTH // 2, SCREEN_HEIGHT // 2)
```

Comment:
There is no need to expose the variable outside this file, hence the name and underscore

Reply:
Exposing it? It’s just the center of the screen. Additionally, you need it in the child classes; you’ve redefined this variable in the child class. I don’t understand why it couldn’t be a normal parameter. I find it much more readable and cleaner.

This translation keeps the original meaning and context intact while ensuring clarity.

Reply:
Ah, I apparently left the original definition in general_menu. That's an oversight

Reply:
> Exposing it? It’s just the center of the screen. Additionally, you need it in the child classes; you’ve redefined this variable in the child class. I don’t understand why it couldn’t be a normal parameter. I find it much more readable and cleaner.
> 
> This translation keeps the original meaning and context intact while ensuring clarity.

The main reason why I don't do this is because every time you setup a menu, you recreate a new tuple with the same data. I feel it's better to keep constant definitions out of function scope

Reply:
Since we only create the menus once, I found it clearer to have it that way. But I think it’s just two different coding styles and viewpoints